.github/workflows/atm-check-queries-run.yml

```diff
@@ -1,13 +1,80 @@
 name: ATM Check Queries Run
 
-# This check is required, therefore we must run it on all PRs, even if only Markdown has changed.
+env:
+  AZURE_STORAGE_URL: "https://atmcodeqldata.blob.core.windows.net"
+  DB_NAME: "AmanSultanBaig/SignIn-SignUp-System-with-Nodejs"
+  DB_PATH: test_db
+  MODEL_PACK_PATH: javascript/ql/experimental/adaptivethreatmodeling/model
+  MODEL_BULDING_PACK_PATH: javascript/ql/experimental/adaptivethreatmodeling/modelbuilding
+  QUERY_SUITE: javascript/ql/experimental/adaptivethreatmodeling/src/codeql-suites/javascript-atm-code-scanning.qls
+
 on:
+  pull_request:
+    paths:
+      - "javascript/ql/experimental/adaptivethreatmodeling/modelbuilding/codeql-pack.lock.yml"
+      - "javascript/ql/experimental/adaptivethreatmodeling/modelbuilding/qlpack.yml"
+      - "javascript/experimental/adpativethreatmodeling/src/qlpack.yml"
+      - "javascript/experimental/adpativethreatmodeling/src/codeql-pack.lock.yml"
   workflow_dispatch:
 
 jobs:
-  hello-world:
+  run-atm-queries:
     runs-on: ubuntu-latest
 
     steps:
-      - name: foo
-        run: echo "Hello world"
+      - uses: actions/checkout@v3
+        with:
+          path: codeql-lib
```

Comment:
This will checkout the repo in the current dir. I don't see how this is desirable, are we sure about this?

Reply:
It's desirable for standardisation and because other Actions can make the assumption that they have a checkout of the repo in the expected place.

Reply:
to make sure I understand: we do ditch this approach for workflows involving several checkouts, right?

Reply:
and does it mean we need to worry about where we create our database? there's no guarantee that `DB_PATH` doesn't collide with a directory from the checkout

Reply:
> to make sure I understand: we do ditch this approach for workflows involving several checkouts, right?

If the workflow needs to check out another repo, we'd typically check out `github/codeql` under the default path and check that the other repo under a different path.

> and does it mean we need to worry about where we create our database? there's no guarantee that DB_PATH doesn't collide with a directory from the checkout

Good point — to avoid this it's good practice to put the database under a temporary directory (see `$RUNNER_TEMP`).

Reply:
The same $RUNNER_TEMP logic really ought to apply to the other checked out repos - or any directory we'd like to create for that matter. I don't have enough knowledge to have a strong opinion but, on the basis of the workflows I interact with, this new approach is a net regression.

Reply:
(I initially thought that the default path would be the name of the repo, similar to the `gh repo clone` command...)

Reply:
> The same `$RUNNER_TEMP` logic really ought to apply to the other checked out repos - or any directory we'd like to create for that matter. I don't have enough knowledge to have a strong opinion but, on the basis of the workflows I interact with, this new approach is a net regression.

The only difficulty with that is it becomes slightly harder to reference where the repo is, so you'd have to write `working-directory: ${{ env.CODEQL_LIB_PATH }}` for example.

I'll have to disagree with you on the net regression.  The Action is now considerably terser and easier to read in my opinion.

Reply:
It's fine to disagree - on this workflow the workflow is actually more verbose and convoluted: instead of checking out in `foo` and then creating a `bar` without worrying we have to use temporary folders. And the more repos we need to checkout / directories to check out the worse it will get. But I admit I don't understand what you mean initially with the expected benefit:

> It's desirable for standardisation and because other Actions can make the assumption that they have a checkout of the repo in the expected place.

Did you mean composite Actions? In the case of this workflow the drawbacks jump out but I'm missing the improvements.

Reply:
(having said that `$RUNNER_TEMP` is very convenient - but the inconsistency of the approach remains in my opinion)

Reply:
> Did you mean composite Actions? In the case of this workflow the drawbacks jump out but I'm missing the improvements.

The improvements are most noticeable when using composite Actions, but I think there's a small benefit in standardisation having the repo checked out in the same place — it reduces cognitive load a tiny bit.